### PR TITLE
Rename proto::SparsePoseGraph.

### PR DESCRIPTION
**Pull request by wohe**
[RFC=0001](https://github.com/googlecartographer/rfcs/blob/master/text/0001-renaming-sparse-pose-graph.md)

### CODE REVIEW

File: cartographer/mapping/pose_graph.h [hunk 1]
```diff
@@ -26,8 +26,8 @@
 #include "cartographer/common/lua_parameter_dictionary.h"
 #include "cartographer/mapping/id.h"
 #include "cartographer/mapping/pose_graph_trimmer.h"
+#include "cartographer/mapping/proto/pose_graph.pb.h"
```
Comment (gaschler) [inline cartographer/mapping/pose_graph.h]: This change will also be necessary in cartographer_ros/cartographer_ros/assets_writer_main.cc and perhaps others on the ros side.

Comment (wohe) [reply cartographer/mapping/pose_graph.h]: Of course. But it makes no sense to send the follow up PR until this is in. And it only needs changes to `assets_writer_main.cc`, no others.

**Comment by wohe** [comment]
@wally-the-cartographer merge

**Comment by wally-the-cartographer** [comment]
Merge requested by authorized user wohe. Merge queue now has a length of 1.

**Pull request merged**